mixer/pkg/attribute/envoyProtoBag.go

```diff
@@ -0,0 +1,316 @@
+		GetAddress()).To16())
+	reqMap["target.ip"] = []byte(net.ParseIP(req.GetAttributes().GetDestination().GetAddress().GetSocketAddress().
+		GetAddress()).To16())
+	reqMap["destination.port"] = int64(req.GetAttributes().GetDestination().GetAddress().GetSocketAddress().
+		GetPortValue())
+	reqMap["destination.principal"] = req.GetAttributes().GetDestination().GetPrincipal()
+	reqMap["source.principal"] = req.GetAttributes().GetSource().GetPrincipal()
+	reqMap["source.ip"] = []byte(net.ParseIP(req.GetAttributes().GetSource().GetAddress().GetSocketAddress().
+		GetAddress()).To16())
+	reqMap["origin.ip"] = []byte(net.ParseIP(req.GetAttributes().GetSource().GetAddress().GetSocketAddress().
+		GetAddress()).To16())
+	reqMap["source.port"] = int64(req.GetAttributes().GetSource().GetAddress().GetSocketAddress().GetPortValue())
+	//reqMap["source.uid"] = ""
+	//reqMap["destination.uid"] = ""
+	reqMap["request.headers"] = req.GetAttributes().GetRequest().GetHttp().GetHeaders()
+	reqMap["request.host"] = req.GetAttributes().GetRequest().GetHttp().GetHost()
+	reqMap["request.method"] = req.GetAttributes().GetRequest().GetHttp().GetMethod()
+	reqMap["request.path"] = req.GetAttributes().GetRequest().GetHttp().GetPath()
+	reqMap["request.scheme"] = req.GetAttributes().GetRequest().GetHttp().GetScheme()
+	reqMap["request.time"] = time.Unix(int64(req.GetAttributes().GetRequest().GetTime().Seconds),
+		int64(req.GetAttributes().GetRequest().GetTime().Nanos))
+	//reqMap["request.url_path"] = "unknown"
+	reqMap["request.useragent"] = req.GetAttributes().GetRequest().GetHttp().GetHeaders()["user-agent"]
+
+
+	pb.reqMap = reqMap
+
+
+
+	scope.Debugf("Returning bag with attributes:\n%v", pb)
+
+	return pb
+}
+
+// GetProtoBag returns a proto-based attribute bag.
+// When you are done using the proto bag, call the Done method to recycle it.
+func GetEnvoyProtoBagAccessLog(msg *accessLogGRPC.StreamAccessLogsMessage, num int) *EnvoyProtoBag {
+	// build the message-level dictionary
+	reqMap := make(map[string]interface{})
+	if httpLogs := msg.GetHttpLogs(); httpLogs != nil {
+		reqMap["context.protocol"] = ""
+		reqMap["destination.ip"] = []byte(net.ParseIP(httpLogs.GetLogEntry()[num].GetCommonProperties().
+			GetDownstreamLocalAddress().GetSocketAddress().GetAddress()).To16())
+		reqMap["destination.port"] = int64(httpLogs.GetLogEntry()[num].GetCommonProperties().
+			GetDownstreamLocalAddress().GetSocketAddress().GetPortValue())
+		reqMap["source.ip"] = []byte(net.ParseIP(httpLogs.GetLogEntry()[num].GetCommonProperties().
+			GetDownstreamRemoteAddress().GetSocketAddress().GetAddress()).To16())
+		reqMap["source.port"] = int64(httpLogs.GetLogEntry()[num].GetCommonProperties().
+			GetDownstreamRemoteAddress().GetSocketAddress().GetPortValue())
+		reqMap["source.namespace"] = httpLogs.GetLogEntry()[num].GetCommonProperties().GetMetadata().GetFilterMetadata()["istio_authn"].GetFields()["source.namespace"].GetStringValue()
```

Comment:
this doesn't feel quite right to me. can we rely on `kubernetesenv` to populate this for us, without relying on `istio_authn` filter metadata?

Reply:
I think so. I'll remove those ones that use Metadata.